packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx

```diff
@@ -76,6 +90,43 @@ class HorizontalOverflow extends Foundation<HorizontalOverflowProps,  React.AllH
         this.setState({
             itemsHeight
         });
+
+        if (canUseDOM() && this.horizontalOverflowItemsRef.current) {
+            this.horizontalOverflowItemsRef.current.addEventListener("scroll", this.throttled);
+        }
+    }
+
+    /**
+     * React life-cycle method
+     */
+    public componentWillUnmount(): void {
+        if (canUseDOM() && this.horizontalOverflowItemsRef.current) {
+            this.horizontalOverflowItemsRef.current.removeEventListener("scroll", this.throttled);
+        }
+    }
+
+    /**
+     * Callback on scroll change
+     */
+    public onScrollChange = (): IScrollChange => {
```

Comment:
why is this public?

Reply:
So consumers can access it...?

Reply:
That is what the callback is for.

Reply:
To put it more illustratively - it's not really the React convention to expose public methods for consumers to call. To do that, you would have to use a `ref` to obtain a reference to the component and imperatively call a function. Also, publicly calling this function would do a consumer no good because this is a function that needs to be called when the scroll changes - not arbitrarily by the consumer.

Instead, it should be a private function used by the class itself and consumers should register a callback (`this.props.onScrollChage`) to handle updates.

Reply:
I guess I am confused - I thought the function had to be public to access - I will change to private.

Reply:
That's the point - no one should be accessing it.